docs/backend-system/building-backends/08-migrating.md

```diff
@@ -98,7 +99,9 @@ import { legacyPlugin } from '@backstage/backend-common';
 
 const backend = createBackend();
 /* highlight-add-next-line */
-backend.add(legacyPlugin('todo', import('./plugins/todo')));
+backend.add(
```

Comment:
mmm you can already do this without changing the signature of the `.add` method.

`backend.add(legacyPlugin('todo', Router().use(yourMiddleware, await todo(todoEnv))))`

I don't think changing the `.add` method is the right way to go, as this is just pure express logic.

Reply:
hmm, that's fair. I didn't want to expose router, but this is probably the easiest way (albeit not documented/mentioned)